### PR TITLE
fix(permissions): Reset custom permissions on promotion

### LINKED ISSUE #9298: Custom permission flag shown for moderators
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Invite a user to a room
2. Grant "all permissions"
3. Promote to moderator

### Expected behaviour
Looks like a normal moderator

### Actual behaviour
Is a moderator but also shows custom permissions icon, but there is no way to reset:
image

### PR DESCRIPTION
**Pull request by nickvergessen**
### ☑️ Resolves

* Fix #9298 

### 🚧 Tasks

- [x] Reset permissions on promotion
- [x] Add a repair step to execute on update

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: appinfo/info.xml [hunk 1]
```diff
@@ -73,6 +73,7 @@ And in the works for the [coming versions](https://github.com/nextcloud/spreed/m
 			<step>OCA\Talk\Migration\CreateHelpCommand</step>
 			<step>OCA\Talk\Migration\ClearResourceAccessCache</step>
 			<step>OCA\Talk\Migration\CacheUserDisplayNames</step>
+			<step>OCA\Talk\Migration\ResetCustomPermissionsForModerators</step>
```
Comment (vitormattos) [inline appinfo/info.xml:76]: Need to be repair step?
Isn't this a pontual change?

Comment (nickvergessen) [reply appinfo/info.xml:76]: yeah, shouldn't happen anymore after the code change, unless we miss something.

Comment (vitormattos) [reply appinfo/info.xml]: Was the other steps created with the same intention to run only a time?

Comment (nickvergessen) [reply appinfo/info.xml]: No, they are "safety nets" to repair undesired admin actions, etc.

**Pull request merged**